scripts/py_matter_yamltests/matter_yamltests/yaml_loader.py

```diff
@@ -0,0 +1,209 @@
+            'busyWaitMs': int,
+            'wait': str,
+        }
+
+        self.__check(content, schema)
+        self.__rule_node_id_and_group_id_are_mutually_exclusive(content)
+        self.__rule_group_step_should_not_expect_a_response(content)
+        self.__rule_step_with_verification_should_be_disabled_or_interactive(
+            content)
+
+        if 'arguments' in content:
+            arguments = content.get('arguments')
+            self.__check_test_step_arguments(arguments)
+
+        if 'response' in content:
+            response = content.get('response')
+            if isinstance(response, list):
+                [self.__check_test_step_response(x) for x in response]
+            else:
+                self.__check_test_step_response(response)
+
+    def __check_test_step_arguments(self, content):
+        schema = {
+            'values': list,
+            'value': (type(None), bool,  str, float, int, dict, list),
+        }
+
+        self.__check(content, schema)
+
+    def __check_test_step_response(self, content):
+        self.__rule_response_value_and_values_are_mutually_exclusive(content)
+
+        if 'values' in content:
+            values = content.get('values')
+            self.__check_type('values', values, list)
+            [self.__check_test_step_response_value(
+                x, allow_name_key=True) for x in values]
+        else:
+            self.__check_test_step_response_value(content)
+
+    def __check_test_step_response_value(self, content, allow_name_key=False):
+        schema = {
+            'value': (type(None), bool, str, int, float, dict, list),
+            'name': str,
+            'error': str,
+            'clusterError': int,
+            'constraints': dict,
+            'saveAs': str
+        }
+
+        if allow_name_key:
+            schema['name'] = str
+
+        self.__check(content, schema)
+
+        if 'constraints' in content:
+            constraints = content.get('constraints')
+            self.__check_test_step_response_value_constraints(constraints)
+
+    def __check_test_step_response_value_constraints(self, content):
+        schema = {
+            'hasValue': bool,
+            'type': str,
+            'minLength': int,
+            'maxLength': int,
+            'isHexString': bool,
+            'startsWith': str,
+            'endsWith': str,
+            'isUpperCase': bool,
+            'isLowerCase': bool,
+            'minValue': (int, float, str),  # Can be a variable
+            'maxValue': (int, float, str),  # Can be a variable
+            'contains': list,
+            'excludes': list,
+            'hasMasksSet': list,
+            'hasMasksClear': list,
+            'notValue': (type(None), bool, str, int, float, list, dict)
+        }
+
+        self.__check(content, schema)
+
+    def __check(self, content, schema):
+        for key in content:
+            if key not in schema:
+                raise KeyError(f'Unknown key: {key}')
```

Comment:
Right, see, propagating that information about "which key" out would be very helpful above....

Reply:
See previous comment.